### PR TITLE
Add code host compatibility and maturity tracking data/pages

### PR DESCRIPTION
**Pull request by jasonyavorsky**
There is an index of links to the two new pages at https://deploy-preview-455--sourcegraph-handbook.netlify.app/product#feature-reference.
~~

This PR adds a set of data files that can be used to autogenerate pages in the handbook/about site that show things like:

- Feature maturity
- Feature hierarchies
- Feature compatibility for code hosts
- Detailed pricing comparisons
- Competitor comparisons
- Probably more creative ideas

It can also serve as the canonical single source of truth for this data in a way that we can build other automation on top of it, as you can see with the new pages linked at the top, which are intended to be an autogenerated version of https://docs.google.com/spreadsheets/d/101JXaau2EPvi322AOFmNeoeuXSJqlruD8gBBsHl1fmI/edit#gid=0.

There really isn't any validation in place except what is done by prettier (which at least makes sure it's valid YAML). That would be a good next iteration. 

I also started a discussion thread on the approach at https://sourcegraph.slack.com/archives/C01SD4A49E1/p1634652195298000.

* this PR originally had a working implementation of pricing tiers, but it was removed because it in theory conflicts with the about site pricing (even though that one doesn't really list specific features, it could create confusion). The commit which reverted that functionality, in case we want to re-add it later, was https://github.com/sourcegraph/handbook/pull/455/commits/fdb000bad2a919b192ecb0d05cc24ac9c7f8c43e.

### CODE REVIEW

**Comment by jasonyavorsky** [comment]
@christinaforney this would supercede the one I made after joining in the old handbook (https://github.com/sourcegraph/about/pull/4617)

File: content/product/index.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ You can reach us at the #product channel or @product-team on Slack. If you have
 
 The product strategy and what we're working on next can be found on the [Sourcegraph strategy page](../company/strategy/index.md). There is also an (internal only) [roadmap presentation](https://docs.google.com/presentation/d/1o3R8WUIhzzRz0x5laTwVcizOzVWrMBe5MCAz74H45Ss/edit#slide=id.gf131fe1596_2_7) available which highlights major features coming next as well as what was recently shipped.
 
+## Feature Maturity
+
+We have an page listing our [features and maturity levels](maturity.md).
```
Comment (jasonyavorsky) [inline content/product/index.md]:
```suggestion
We have a page listing our [features and maturity levels](maturity.md).
```

File: README.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ Then start the handbook:
 yarn dev
 ```
 
+## Autogenerated pages
+
+There is an autogenerated page at `content/product/maturity.md` that is created at build time from the YAML files in the [data](./data) folder. The script which does this is in [src/scripts/generated-pages.mjs](./src/scripts/generated-pages.mjs).
```
Comment (jasonyavorsky) [inline README.md]:
```suggestion
There are autogenerated pages at `content/product/maturity.md` and `content/product/features_by_tier.md` that are created at build time from the YAML files in the [data](./data) folder. The script which does this is in [src/scripts/generated-pages.mjs](./src/scripts/generated-pages.mjs).
```

**Comment by Joelkw** [comment]
Thanks for moving this forward! Left comments[ in the slack thread](https://sourcegraph.slack.com/archives/C01SD4A49E1/p1634952093028200?thread_ts=1634652195.298000&cid=C01SD4A49E1)!

File: README.md [hunk 3]
```diff
@@ -22,6 +22,10 @@ Then start the handbook:
 yarn dev
 ```
 
+## Autogenerated pages
+
+There are autogenerated pages at `content/product/maturity.md` and `content/product/features_by_tier.md` that are created at build time from the YAML files in the [data](./data) folder. The script which does this is in [src/scripts/generated-pages.mjs](./src/scripts/generated-pages.mjs).
```
Comment (jasonyavorsky) [inline README.md]:
```suggestion
There are autogenerated pages in `content/product/` that are created at build time from the YAML files in the [data](./data) folder. The script which does this is in [src/scripts/generated-pages.mjs](./src/scripts/generated-pages.mjs).
```

File: content/product/index.md [hunk 4]
```diff
@@ -15,6 +15,14 @@ You can reach us at the #product channel or @product-team on Slack. If you have
 
 The product strategy and what we're working on next can be found on the [Sourcegraph strategy page](../company/strategy/index.md). There is also an (internal only) [roadmap presentation](https://docs.google.com/presentation/d/1o3R8WUIhzzRz0x5laTwVcizOzVWrMBe5MCAz74H45Ss/edit#slide=id.gf131fe1596_2_7) available which highlights major features coming next as well as what was recently shipped.
 
+## Feature reference
+
+We have a several feature matrix pages:
```
Comment (jasonyavorsky) [inline content/product/index.md]:
```suggestion
We have a several feature reference matrices:
```

File: content/company/strategy/index.md [hunk 5]
```diff
@@ -81,6 +81,10 @@ Additionally, our [product design principles](../../product/design/product_desig
 - If your organization is getting value from Sourcegraph with a lot of users, our [pricing](https://about.sourcegraph.com/pricing) is designed so that we earn money from you. This lets us invest in improving our product.
 - All users at a given customer are on the same pricing tier. This is simpler than having users at different tiers and encourages us to build things that are broadly valuable.
 
+## Where we're at now
+
+A complete list of feature areas by maturity, tier, or by code host compatibility can be found on [our feature reference page](../../product/index.md#feature-reference).
```
Comment (jasonyavorsky) [inline content/company/strategy/index.md]:
```suggestion
A complete list of feature areas by maturity, tier, or by code host compatibility can be found on [our feature reference pages](../../product/index.md#feature-reference).
```

File: content/product/index.md [hunk 6]
```diff
@@ -15,6 +15,14 @@ You can reach us at the #product channel or @product-team on Slack. If you have
 
 The product strategy and what we're working on next can be found on the [Sourcegraph strategy page](../company/strategy/index.md). There is also an (internal only) [roadmap presentation](https://docs.google.com/presentation/d/1o3R8WUIhzzRz0x5laTwVcizOzVWrMBe5MCAz74H45Ss/edit#slide=id.gf131fe1596_2_7) available which highlights major features coming next as well as what was recently shipped.
 
+## Feature reference
+
+We have a several feature reference matrices:
```
Comment (jasonyavorsky) [inline content/product/index.md]:
```suggestion
We have several feature reference matrices:
```

File: content/product/product_management/index.md [hunk 7]
```diff
@@ -21,6 +21,10 @@ We also use [per-area strategy pages](../../company/strategy/index.md#per-area-s
 
 All product engineering teams should have their own strategy page, and should update it on a monthly cadence (at the same time the [PMM roadmap](https://docs.google.com/presentation/d/1o3R8WUIhzzRz0x5laTwVcizOzVWrMBe5MCAz74H45Ss/edit#slide=id.gf131fe1596_2_7) (internal only) is updated) by proposing a new pull request to update their content. Everyone on the team should be invited to contribute, and the update should be shared with important stakeholders. This helps us keep everyone aligned, and ensures we have a single source of truth for our product directions. The specific day of the month that the team updates the doc is up to the team internally to decide, the important part is that it is never more than a month out of date.
 
+### Feature data
+
+We have a several [feature matrix pages](../index.md#feature-reference) that are backed by data in the [data folder of the handbook project](https://github.com/sourcegraph/handbook). Product managers should keep these files up to date as part of their monthly direction page updates, as needed.
```
Comment (jasonyavorsky) [inline content/product/product_management/index.md]:
```suggestion
We have several [feature matrix pages](../index.md#feature-reference) that are backed by data in the [data folder of the handbook project](https://github.com/sourcegraph/handbook). Product managers should keep these files up to date as part of their monthly direction page updates, as needed.
```

File: data/features.yml [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+  product_area: search
+  maturity: not_implemented
+  documentation_link: false
+  compatibility: false
+
+saved_searches:
+  title: 'Saved Searches'
+  description: ''
+  product_area: search
+  maturity: deprecated
```
Comment (limitedmage) [inline data/features.yml:133]: Saved searches are not deprecated. Only *notifications* for saved searches are deprecated (in favor of Code Monitoring).

**Pull request merged**